### PR TITLE
fixing FPN parsing

### PR DESCRIPTION
**Pull request by makrsmark**
also supporting more variants

### CODE REVIEW

File: lib/utils/flight_plan_utils.ts [hunk 1]
```diff
@@ -105,23 +111,29 @@ export class FlightPlanUtils {
...
         decodeResult.raw.route_status = 'RP';
         decodeResult.formatted.items.push({
           type: 'status',
           code: 'ROUTE_STATUS',
           label: 'Route Status',
           value: 'Route Planned',
         });
+        if (fields[i].length > 2) {
+          allKnownFields = allKnownFields && this.processFlightPlan(decodeResult, fields[i].split(':'));
+        }
```
Comment (makrsmark) [inline lib/utils/flight_plan_utils.ts:124]: yes - this is now recursive

File: lib/utils/flight_plan_utils.ts [hunk 2]
```diff
@@ -105,23 +111,29 @@ export class FlightPlanUtils {
...
         decodeResult.raw.route_status = 'RI';
         decodeResult.formatted.items.push({
           type: 'status',
           code: 'ROUTE_STATUS',
           label: 'Route Status',
           value: 'Route Inactive',
         });
+        if (fields[i].length > 2) {
+          allKnownFields = allKnownFields && this.processFlightPlan(decodeResult, fields[i].split(':'));
+        }
```
Comment (makrsmark) [inline lib/utils/flight_plan_utils.ts:136]: also another recursive entry point

**Review by kevinelliott — APPROVED** [review]
LGTM, thanks @makrsmark !

File: lib/utils/flight_plan_utils.ts [hunk 3]
```diff
@@ -105,23 +111,29 @@ export class FlightPlanUtils {
         if (fields[i].length > 2) {
           addRoute(decodeResult, fields[i].substring(2));
```
Comment (makrsmark) [inline lib/utils/flight_plan_utils.ts:112]: should this follow the other ones for consistency?

Comment (kevinelliott) [reply lib/utils/flight_plan_utils.ts:112]: Yes, and eventually we can extract out routing functions, etc.

Comment (makrsmark) [reply lib/utils/flight_plan_utils.ts:112]: no - it goes right into the route without a colon

**Pull request merged**